mlflow/store/artifact/mlflow_artifacts_repo.py

```diff
@@ -0,0 +1,25 @@
+from urllib import parse
+from collections import namedtuple
+from mlflow.store.artifact.http_artifact_repo import HttpArtifactRepository
+from mlflow.tracking._tracking_service.utils import get_tracking_uri
+
+
+def _resolve_connection_params(artifact_uri):
+    ParsedURI = namedtuple("ParsedURI", "scheme host port path")
+    parsed_uri = parse.urlparse(artifact_uri)
+    return ParsedURI(parsed_uri.scheme, parsed_uri.hostname, parsed_uri.port, parsed_uri.path)
+
+
+class MlflowArtifactsRepository(HttpArtifactRepository):
+    """Scheme wrapper around HttpArtifactRepository for mlflow-artifacts server functionality"""
+
+    def __init__(self, artifact_uri):
+        parsed = _resolve_connection_params(artifact_uri)
+        tracking_uri = get_tracking_uri()
+        resolved_artifacts_uri = (
+            artifact_uri.replace("mlflow-artifacts:", f"{tracking_uri}")
+            .replace(f"mlflow-artifacts:{parsed.host}", f"{tracking_uri}")
+            .replace(f"mlflow-artifacts:{parsed.host}:{parsed.port}", f"{tracking_uri}")
```

Comment:
We should only swap in the tracking URI if the artifact URI doesn't specify a hostname.

Reply:
Just to confirm, if `artifact_uri` is `mlflow-artifacts:/my/artifact/path`, it should be convert to `<tracking uri>/api/2.0/mlflow-artifacts/artifacts/my/artifact/path`, correct?

Reply:
Yep! I added some additional parameter tests for the mock harness to validate the user-defined pathing following the scheme patch.